pkg/boot/kexec/kexec_linux_implemented.go

```diff
@@ -26,8 +34,38 @@ func FileLoad(kernel, ramfs *os.File, cmdline string) error {
 		flags |= unix.KEXEC_FILE_NO_INITRAMFS
 	}
 
+	if isGzip(kernel) {
+		g, err := gzip.NewReader(kernel)
+		if err != nil {
+			return fmt.Errorf("could read kernel gzip header: %v", err)
+		}
+		decompressed, err := ioutil.TempFile("", "kernel")
+		if err != nil {
+			return fmt.Errorf("could not create temp file: %v", err)
+		}
+		defer os.Remove(decompressed.Name())
+		if _, err := io.Copy(decompressed, g); err != nil {
+			return fmt.Errorf("could not decompress kernel: %v", err)
+		}
+		if err := g.Close(); err != nil {
+			return fmt.Errorf("could not close gunzip (bad checksum?): %v", err)
+		}
+		if err := decompressed.Sync(); err != nil {
+			return fmt.Errorf("could not sync decompressed kernel: %v", err)
+		}
```

Comment:
printing the name "might" be helpful as knowing the path might indicate the problem, e.g. /tmp was out of room so the sync failed.